system-contracts/contracts/precompiles/EcAdd.yul

```diff
@@ -413,28 +441,35 @@ object "EcAdd" {
 
             // P1 + P2 = P3
 
-            x1 := intoMontgomeryForm(x1)
-            y1 := intoMontgomeryForm(y1)
-            x2 := intoMontgomeryForm(x2)
-            y2 := intoMontgomeryForm(y2)
+            m_x1 := intoMontgomeryForm(x1)
+            m_y1 := intoMontgomeryForm(y1)
+            m_x2 := intoMontgomeryForm(x2)
+            m_y2 := intoMontgomeryForm(y2)
 
             // Ensure that the points are in the curve (Y^2 = X^3 + 3).
-            if or(iszero(pointIsInCurve(x1, y1)), iszero(pointIsInCurve(x2, y2))) {
+            if or(iszero(pointIsInCurve(m_x1, m_y1)), iszero(pointIsInCurve(m_x2, m_y2))) {
                 burnGas()
             }
 
-            // (y2 - y1) / (x2 - x1)
-            let slope := montgomeryDiv(submod(y2, y1, P()), submod(x2, x1, P()))
-            // x3 = slope^2 - x1 - x2
-            let x3 := submod(montgomeryMul(slope, slope), addmod(x1, x2, P()), P())
-            // y3 = slope * (x1 - x3) - y1
-            let y3 := submod(montgomeryMul(slope, submod(x1, x3, P())), y1, P())
-
-            x3 := outOfMontgomeryForm(x3)
-            y3 := outOfMontgomeryForm(y3)
+            mstore(0, x1)
```

Comment:
Does the circuit check that x1/y1/x2/y2 
1) are less than modulo? 
2) are on the curve?

Reply:
We have these checks in yul contract being a vanguard call, therefore there are no additional checks in-circuit because these will duplicate the logic leading to constraints amount improvement and performance degradation